src/ray/raylet/worker_pool.cc

```diff
@@ -664,15 +664,23 @@ void WorkerPool::PushWorker(const std::shared_ptr<WorkerInterface> &worker) {
   RAY_CHECK(worker->GetAssignedTaskId().IsNil())
       << "Idle workers cannot have an assigned task ID";
   auto &state = GetStateForLanguage(worker->GetLanguage());
-  auto it = state.dedicated_workers_to_tasks.find(worker->GetProcess());
-  if (it != state.dedicated_workers_to_tasks.end()) {
-    // The worker is used for the actor creation task with dynamic options.
-    // Put it into idle dedicated worker pool.
-    const auto task_id = it->second;
+  auto it_p = state.pending_dedicated_workers_to_tasks.find(worker->GetProcess());
+  auto it_r = state.registered_dedicated_workers_to_tasks.find(worker->GetProcess());
```

Comment:
It seems like we should be able to combine these into a single map, but I'm probably missing something -- what's the reason they need to be separate?

Reply:
Good point, at first I thought this wasn't possible but I think I was limiting myself by trying to make an incremental change.  Let me rewrite things so it just uses one map